src/main/java/uk/gov/ons/ctp/response/casesvc/service/impl/CaseServiceImpl.java

```diff
@@ -185,21 +188,76 @@ public CaseEvent createCaseEvent(CaseEvent caseEvent, Case newCase, Timestamp ti
       createAdHocAction(category, caseEvent);
 
       // should a new case be created?
-      createNewCase(category, caseEvent, targetCase, newCase);
+      //createNewCase(category, caseEvent, targetCase, newCase);
 
       // transition case group status
       transitionCaseGroupStatus(caseEvent, targetCase);
+
+      // if this is a respondent enrolling event
+      if (caseEvent.getCategory().toString().equals("RESPONDENT_ENROLED")) {
+        // are there other case groups that need updating
+        transitionOtherCaseGroups(category, caseEvent, targetCase, newCase);
+      } else {
+        createNewCase(category, caseEvent, targetCase, newCase);
+        effectTargetCaseStateTransition(category, targetCase);
+      }
     }
 
     return createdCaseEvent;
   }
 
+  /**
+   * Use case ID to find case group
+   * Use case group to find collectionEx
+   * Then use CollEx service to find survey ID, then find all collexs for survey
+   * Then get all casge groups for the party ID where Collex ID is in list of collexs
+   * TODO: null checking
+   */
+  private void transitionOtherCaseGroups(final Category category, final CaseEvent caseEvent, final Case targetCase, Case newCase) throws CTPException {
+    CaseGroup caseGroup = caseGroupRepo.findOne(targetCase.getCaseGroupFK());
+    CollectionExerciseDTO collectionExercise = collectionExerciseSvcClientService
+            .getCollectionExercise(caseGroup.getCollectionExerciseId());
+    // fetch all the collection exercises for a survey
+    List<CollectionExerciseDTO> collectionExercises = collectionExerciseSvcClientService.getCollectionExercises(collectionExercise.getSurveyId());
+    // get published collection exercise
+    List<CollectionExerciseDTO> publishedCollexs = collectionExercises.stream().filter(ce -> ce.getState().toString().equals("READY_FOR_LIVE")).collect(Collectors.toList());
+    // get list of collection exercise ids
+    List<UUID> collExs = publishedCollexs.stream().map(CollectionExerciseDTO::getId).collect(Collectors.toList());
+    // fetch party ID for the RU
+    UUID partyId = targetCase.getPartyId();
+    //select * from casesvc.casegroup where partyid = partyId and collectionexerciseid in collectionExercises
+    List<CaseGroup> caseGroups = caseGroupRepo.retrieveByPartyIdInListOfCollEx(partyId, collExs);
+    checkCaseState(category, caseGroups, caseEvent, newCase);
+  }
+
+  /**
+   * This has been triggered by a 'RESPONDENT_ENROLED' event. If we find any associated Case Groups
+   * with only B cases we need to make case 'INACTIONABLE' and create BI case.
+   */
+  private void checkCaseState(final Category category, List<CaseGroup> caseGroups, CaseEvent caseEvent, Case newCase) throws CTPException {
+    // check all case groups are type B. For surveys this is always true
+    List<CaseGroup> caseGroupsToUpdate = caseGroups.stream().filter(cg -> cg.getSampleUnitType().toString().equals("B")).collect(Collectors.toList());
```

Comment:
the `toString()` in this line is not required

Reply:
I think it is to get a proper String comparison.